### PR TITLE
Dynamic installation of vSphere IPI cluster

### PR DESCRIPTION
**Pull request by vavuthu**
This PR will lift the limitation of deploying single cluster in vSphere IPI. Below are important changes with this PR

1. Lifted the limitation of deploying 1 cluster, with this PR we can deploy n number of clusters based on our VMware resources
2. Dynamic creation and deletion of DNS records
3. Dynamic allocation and releasing of IP's
4. Changed base domain to qe.rh-ocs.com
5. vSphere IPI is deployed with the same cluster name as we passed as parameter to ocs-ci

Important commits with this PR:

d55e00e87fdf1abec4562ccd176051ef2ce39a79 : Adding IPAM module
ca65de0e4861a9084e7b7f8dd466dd82608ee339 : fetching hosted zone ID and record sets for domain in aws
b9bb4fb96c68b72704c9f031391b1dd4f3c45f05 : Assigning IPs in IPAM server and creating DNS records in aws
4e7ac5c8ba2709863ce658e5b5d1f2adcecdd272 : Delete DNS records and release IP's from IPAM server
a64d5e2704a3c51c97e8530c5e0a0dea4d7c306f : Changing base domain to qe.rh-ocs.com and cluster name to sync with ocs-ci cluster name parameter                                                              

Signed-off-by: vavuthu <vavuthu@redhat.com>

### CODE REVIEW

File: ocs_ci/utility/ipam.py [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+        Args:
+            hostname (str): Hostname to release IP
+
+        """
+        # release the IP
+        endpoint = os.path.join("http://", self.ipam, "api/removeHost.php?")
+        payload = {"apiapp": self.apiapp, "apitoken": self.token, "host": hostname}
+        res = requests.post(endpoint, data=payload)
+        if res.status_code == "200":
+            logger.info(f"Successfully released {hostname} IP from IPAM server")
```
Comment (vasukulkarni) [inline ocs_ci/utility/ipam.py:109]: how about adding else: with exception("Failed to update IPAM") and using retry decorator

```
@retry(IPAMUpdateFailed, tries=5, delay=3, backoff=1)
```

Comment (vavuthu) [reply ocs_ci/utility/ipam.py:115]: Done in latest commit

**Review by petr-balogh — APPROVED** [review]

File: ocs_ci/utility/ipam.py [hunk 2]
```diff
@@ -0,0 +1,130 @@
...
+        except requests.exceptions.Timeout as timeout_err:
+            logger.error(f"Timeout Error: {timeout_err}")
+            raise
+        except requests.exceptions.RequestException as err:
+            logger.error(f"Unexpected Error: {err}")
+            raise
+
+        if "Error" in res.text:
+            logger.error(f"Error in assigning IP to host. Error: {res.text}")
+            raise
```
Comment (dahorak) [inline ocs_ci/utility/ipam.py:79]: I'm not sure, if I didn't miss something, but which exception is going to raise this `raise` statement?

Comment (vavuthu) [reply ocs_ci/utility/ipam.py]: updated

**Review by petr-balogh — APPROVED** [review]

**Review by dahorak — APPROVED** [review]

**Pull request merged**